### PR TITLE
Fix for options clear upon translation

### PR DESCRIPTION
**Pull request by RensDofferhoff**
Fixes: https://github.com/jasp-stats/INTERNAL-jasp/issues/2262

Its not a nice solution but its pragmatic.
A nice fix would be quite involved due to a delayed signal in description.cpp that other parts seem to require.

### CODE REVIEW

File: Desktop/analysis/analyses.cpp [hunk 1]
```diff
@@ -339,8 +339,9 @@ void Analyses::rescanAnalysisEntriesOfDynamicModule(Modules::DynamicModule * mod
 			{
 				Analysis * a = keyval.second;
 
-				if(a->readyToCreateForm())
+				if(a->readyToCreateForm() && !a->justTranslated())
```
Comment (boutinb) [inline Desktop/analysis/analyses.cpp]: I'm not sure to understand this fix... I see a risk that the form is not created at all...

**Review by boutinb — COMMENTED** [review]
You have to put more explanation here: is the `rescanAnalysisEntriesOfDynamicModule` function called several times? If there is a race issue, which one is it?

**Comment by RensDofferhoff** [comment]
> You have to put more explanation here: is the `rescanAnalysisEntriesOfDynamicModule` function called several times? If there is a race issue, which one is it?

When _qml->retranslate(); is called eventually rescanAnalysisEntriesOfDynamicModule is called exactly once after a delay of 300ms and a bunch of signals around the alteration of the description.
A new form is created and options are cleared.

There is no nicer solution due to the 300 ms delay. You cant store and load the option state or clear a boolean upon the currentLanguageChanged signal because this signal is emitted way before rescanAnalysisEntriesOfDynamicModule is called.

The only way to get a clean solution is to get rid of the 300 ms delay but other parts seem to depend on it.
Therefore we set a boolean upon prepareForLanguageChange signal and clear it in rescanAnalysisEntriesOfDynamicModule.

File: Desktop/analysis/analysis.h [hunk 2]
```diff
@@ -241,7 +242,8 @@ public slots:
 	bool						_isDuplicate					= false,
 								_wasUpgraded					= false,
 								_tryToFixNotes					= false,
-								_hasReport						= false;
+								_hasReport						= false,
+	_aboutToBeTranslated					= false;
```
Comment (JorisGoosen) [inline Desktop/analysis/analysis.h]:
```suggestion
								_aboutToBeTranslated					= false;
```

File: Desktop/analysis/analyses.cpp (hunk 1 above)
Comment (JorisGoosen) [reply Desktop/analysis/analyses.cpp]: I dont really understand why the form would not be there?
Is `checkAnalysisEntry()` returning `false`?

**Comment by JorisGoosen** [comment]
> A new form is created and options are cleared.

Where did the old one go?

File: Desktop/analysis/analyses.cpp (hunk 1 above)
Comment (RensDofferhoff) [reply Desktop/analysis/analyses.cpp]: Its returning true.  therefore we reach the else statement

**Comment by RensDofferhoff** [comment]
> > A new form is created and options are cleared.
> 
> Where did the old one go?

its destroyed in createForm()

**Review by boutinb — APPROVED** [review]

**Pull request merged**